akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -137,6 +143,91 @@ object GraphStageLogic {
   private object DoNothing extends (() ⇒ Unit) {
     def apply(): Unit = ()
   }
+
+  trait MinimalDeathWatch {
+    this: MinimalActorRef ⇒
+    private var watchedBy = ActorCell.emptyActorRefSet
```

Comment:
This is not going to work: watching/unwatching happens concurrently from the outside, this needs to be threadsafe.